app/controllers/doorkeeper_applications_controller.rb

```diff
@@ -17,6 +17,11 @@ def update
     end
   end
 
+  def users_with_access
+    authorized_users_user_ids = @application.authorized_tokens.select(:resource_owner_id)
+    @users = policy_scope(User).where(id: authorized_users_user_ids).includes(:organisation).page(params[:page]).per(100)
```

Comment:
@jamiecobbett this wasn't actually that complicated. See https://github.com/alphagov/signonotron2/commit/0c18c1b8a31f72d107db1f2601cb346c09c67bc4.